vdj_ann/src/annotate.rs

```diff
@@ -145,10 +145,50 @@ pub fn chain_type(b: &DnaString, rkmers_plus_full_20: &[(Kmer20, i32, i32)], rty
 // The structure of the output is:
 // { ( start on sequence, match length, ref tig, start on ref tig, mismatches on sequence ) }.
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd)]
+/// Annotation object denotes matches of a DnaString sequence to a reference sequence
+pub struct Annotation {
+    /// Start position on sequence (contig)
+    pub tig_start: i32,
+    /// Match length
+    pub match_len: i32,
+    /// Reference id (that this contig matches)
+    pub ref_id: i32,
+    /// Start position on reference
+    pub ref_start: i32,
+    /// Number of mismatches on the sequence (contig)
+    pub mismatches: i32,
+}
+
+#[derive(Ord, PartialOrd, PartialEq, Eq)]
+/// Used prior to creatrion of Annotation object, includes a vector of mismatches
+pub struct PreAnnotation {
+    /// Start position on sequence (contig)
+    pub tig_start: i32,
+    /// Match length
+    pub match_len: i32,
+    /// Reference id (that this contig matches)
+    pub ref_id: i32,
+    /// Start position on reference
+    pub ref_start: i32,
+    /// Vector mismatches on the sequence (contig)
+    pub mismatches: Vec<i32>,
+}
+
+pub fn next_diff_pre_annotation(x: &[PreAnnotation], i: i32) -> i32 {
```

Comment:
Same here - always default to making things private to keep the public API of a module to the minimum.
```suggestion
fn next_diff_pre_annotation(x: &[PreAnnotation], i: i32) -> i32 {
```